src/CoreWlan/CWKeychain.cs

```diff
@@ -0,0 +1,231 @@
+		static extern OSStatus CWKeychainCopyWiFiEAPIdentity (CWKeychainDomain domain, NSDataRef ssid, out SecIdentityRef identity);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool FindWiFiEAPIdentity (CWKeychainDomain domain, NSData ssid, out SecIdentity? identity)
+		{
+			identity = null;
+			IntPtr outPtr = IntPtr.Zero;
+			var result = CWKeychainCopyWiFiEAPIdentity (domain, ssid.GetHandle (), out outPtr);
+			if (result == 0)
+			{
+				identity = new SecIdentity (outPtr, true);
+			}
+
+			return result == 0;
+		}
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		[DllImport (Constants.CoreWlanLibrary)]
+		static extern OSStatus CWKeychainDeleteWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSDataRef ssid);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TryDeleteWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSData ssid)
+			=> CWKeychainDeleteWiFiEAPUsernameAndPassword (domain, ssid.GetHandle ()) == 0;
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		[DllImport (Constants.CoreWlanLibrary)]
+		static extern OSStatus CWKeychainDeleteWiFiPassword (CWKeychainDomain domain, NSDataRef ssid);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TryDeleteWiFiPassword (CWKeychainDomain domain, NSData ssid)
+			=> CWKeychainDeleteWiFiPassword (domain, ssid.GetHandle ()) == 0;
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		[DllImport (Constants.CoreWlanLibrary)]
+		static extern OSStatus CWKeychainFindWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSDataRef ssid, out NSStringRef username, out NSStringRef password);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TryFindWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSData ssid, out NSString? username, out NSString? password)
+		{
+			username = null;
+			password = null;
+			NSStringRef usernamePtr = IntPtr.Zero;
+			NSStringRef passwordPtr = IntPtr.Zero;
+			var result = CWKeychainFindWiFiEAPUsernameAndPassword (domain, ssid.GetHandle (), out usernamePtr, out passwordPtr);
+			if (usernamePtr != IntPtr.Zero) {
+				username = Runtime.GetNSObject<NSString> (usernamePtr, true);
+			}
+			if (passwordPtr != IntPtr.Zero) {
+				password= Runtime.GetNSObject<NSString> (passwordPtr, true);
+			}
+			return result == 0;
+		}
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		[DllImport (Constants.CoreWlanLibrary)]
+		static extern OSStatus CWKeychainFindWiFiPassword (CWKeychainDomain domain, NSDataRef ssid, out NSStringRef password);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TryFindWiFiPassword (CWKeychainDomain domain, NSData ssid, out NSString? password)
```

Comment:
Same, `string` output:

```suggestion
		public static bool TryFindWiFiPassword (CWKeychainDomain domain, NSData ssid, out string? password)
```